src/optimizer/query_graph.c

```diff
@@ -1494,13 +1494,11 @@ qo_insert_segment (QO_NODE * head, QO_NODE * tail, PT_NODE * node, QO_ENV * env,
   QO_SEG_HEAD (seg) = head;
   QO_SEG_TAIL (seg) = tail;
   QO_SEG_IDX (seg) = env->nsegs;
-  /* add dummy name to segment example: dummy attr from view transfrom select count(*) from v select count(*) from
-   * (select {v}, 1 from t) v (v, 1) here, '1' is dummy attr set empty string to avoid core crash
-   */
   if (node)
     {
+      /* If it is not PT_NAME, an empty string is used for name. */
       QO_SEG_NAME (seg) =
-	node->info.name.original ? node->info.name.original : pt_append_string (QO_ENV_PARSER (env), NULL, "");
+	node->node_type == PT_NAME ? node->info.name.original : pt_append_string (QO_ENV_PARSER (env), NULL, "");
```

Comment:
Unlike before, is there no need to check the possibility that node->info.name.original is null?
Shouldn't it be something like the following?
`(node->node_type == PT_NAME && node->info.name.original) ? node->info.name.original : pt_append_string (QO_ENV_PARSER (env), NULL, "");`

Reply:
I think the previous check was wrong. But adding a null check is fine. I'll fix it.